### PR TITLE
Add setting to choose invoice prefix (defaults to Civi-)

### PR DESCRIPTION
**Pull request by ananelson**

### CODE REVIEW

**Comment by ananelson** [comment]
This also standardizes the remainder of the invoice number to ensure proper sorting by prepending zeros.

File: CRM/Civiquickbooks/Invoice.php [hunk 1]
```diff
@@ -658,7 +663,7 @@ protected function mapToAccounts($db_contribution, $accountsID, $SyncToken, $qb_
       }
 
       if ($whereToGetInvoiceNumber == 'civi') {
-        $new_invoice['DocNumber'] = 'Civi-' . $db_contribution['id'];
+        $new_invoice['DocNumber'] = $invoice_prefix . sprintf("%07d", $db_contribution['id']);
```
Comment (agileware-fj) [inline CRM/Civiquickbooks/Invoice.php:666]: Alternative would be to do `sprintf($invoice_prefix, $db_contribution['id'])` where `$invoice_prefix` is in the format `'Civi-%07d'` (for this example)

File: settings/civiquickbooks.setting.php [hunk 2]
```diff
@@ -139,6 +139,23 @@
     ),
     'quick_form_type' => 'Element',
   ),
+  'quickbooks_invoice_prefix' => array(
+    'group_name' => E::ts('QuickBooks Online Settings'),
+    'group' => 'civiquickbooks',
+    'name' => 'quickbooks_invoice_prefix',
+    'type' => 'String',
+    'default' => 'Civi-',
```
Comment (agileware-fj) [inline settings/civiquickbooks.setting.php:147]: `'Civi-%07d'` to use format

**Review by agileware-fj — CHANGES_REQUESTED** [review]
The concept behind this is sound, and I'd merge immediately if it wasn't for enforcing the 7 digits of invoice number.

I don't think any of our own clients have breached the 100,000 invoice mark, which makes this extra two (often three) digits awkward in their invoices. It seems like QBO should be capable of sorting numerically on the invoice field anyway, although I haven't tested this yet. In this case is this for offline sorting tools?
Edit: And of course, on testing, it seems like QBO in fact does **not** sort numerically on the invoice number field, but by character.

I think it would be better then to be able to specify a format with a reasonable default, possibly even just `'Civi-%u'` so that the invoice number remains better readable to humans.

Note that this will also require validation to be added to the settings form to make sure the invoice prefix format us usable, e.g.
`strpos(sprintf($invoice_id_format, 99999), '99999') !== FALSE)`.

@ananelson do you have any thoughts on this, and can you update your PR accordingly? (Otherwise will adjust as required and merge when I have time)

**Comment by agileware-fj** [comment]
Rebased, applied suggested edit (default will remain the same), and merged.

Thanks @ananelson for starting this enhancement

**Pull request closed without merging**